Gems/Atom/Feature/Common/Code/Source/CoreLights/LightCommon.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+
+#include <Atom/Feature/CoreLights/LightCommon.h>
+
+namespace AZ::Render::LightCommon
+{
+    bool HasGPUCulling(
+        RPI::Scene* parentScene,
+        const RPI::ViewPtr& view,
+        AZStd::unordered_set<AZStd::pair<const RPI::RenderPipeline*, const RPI::View*>>& gpuCullingData)
+    {
+        for (const auto& renderPipeline : parentScene->GetRenderPipelines())
+        {
+            if (renderPipeline->NeedsRender() && gpuCullingData.contains(AZStd::pair(renderPipeline.get(), view.get())))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    void CacheGPUCullingPipelineInfo(
+        RPI::RenderPipeline* renderPipeline,
+        RPI::ViewPtr newView,
+        RPI::ViewPtr previousView,
+        AZStd::unordered_set<AZStd::pair<const RPI::RenderPipeline*, const RPI::View*>>& gpuCullingData)
+    {
+        // Check if render pipeline is using GPU culling
+        if (!renderPipeline->FindFirstPass(AZ::Name("LightCullingPass")))
+        {
+            return;
+        }
+
+        if (previousView)
+        {
+            gpuCullingData.erase(AZStd::make_pair(renderPipeline, previousView.get()));
+        }
+
+        if (newView)
+        {
+            gpuCullingData.insert(AZStd::make_pair(renderPipeline, newView.get()));
+        }
+    }
+
+    GpuBufferHandler& GetOrCreateVisibleBuffer(
+        uint32_t& visibleBufferUsedCount,
+        AZStd::vector<GpuBufferHandler>& visibleBufferHandlers,
+        const AZStd::string_view& bufferName,
+        const AZStd::string_view& bufferSrgName,
+        const AZStd::string_view& elementCountSrgName)
+    {
+        while (visibleBufferUsedCount >= visibleBufferHandlers.size())
+        {
+            GpuBufferHandler::Descriptor desc;
+            desc.m_bufferName = bufferName;
+            desc.m_bufferSrgName = bufferSrgName;
+            desc.m_elementCountSrgName = elementCountSrgName;
+            desc.m_elementFormat = AZ::RHI::Format::R32_UINT;
+            desc.m_srgLayout = RPI::RPISystemInterface::Get()->GetViewSrgLayout().get();
+
+            visibleBufferHandlers.emplace_back(desc);
+        }
```

Comment:
The implementation suggested if may create more than one buffer. For example, if visibleBufferUsedCount = 3 and visibleBufferHandlers.size() = 1, then it will add 3 more GPUBufferHandler here.
This doesn't look right.

Reply:
Just to clarify. Most the of the changes here is moving the finctions to a common place for utility as it is used by Decals and simple lights. The While loops allows us to create multiple gpu buffers in case we want to cull stuff for multiple views in a frame.

Reply:
The comment at the function's declaration said "// Get or create a buffer that will be used for visibility when doing CPU culling."
Which suggests the function may create a buffer. 

Also the current comment didn't mention about the input parameters. 
It's easy to understand that last three parameters. But it's not easy for user to understand the first two parameters and why.

Reply:
Updated the implementation to address the feedback.